src/commands/use.ts

```diff
@@ -5,67 +5,86 @@ import * as semver from 'semver'
-        this.debug(`Flag overriden target channel: ${this.channel}`)
-      }
+    // Determine if the version is from a different channel and update to account for it (ex. cli-example update 3.0.0-next.22 should update the channel to next as well.)
+    const versionParts = targetVersion?.split('-') || ['', '']
+    if (versionParts && versionParts[1]) {
+      this.channel = versionParts[1].substr(0, versionParts[1].indexOf('.'))
+      this.debug(`Flag overriden target channel: ${this.channel}`)
+    }
 
-      await this.ensureClientDir()
-      this.debug(`Looking for locally installed versions at ${this.clientRoot}`)
-
-      // Do not show known non-local version folder names, bin and current.
-      const versions = fs.readdirSync(this.clientRoot).filter(dirOrFile => dirOrFile !== 'bin' && dirOrFile !== 'current')
-      if (versions.length === 0) throw new Error('No locally installed versions found.')
-
-      if (versions.includes(targetVersion)) {
-        this.updateToExistingVersion(targetVersion)
-      } else if (channelUpdateRequested) {
-        // Begin prompt
-        cli.action.start(`${this.config.name}: Updating CLI`)
-
-        // Run pre-update hook
-        await this.config.runHook('preupdate', {channel: this.channel})
-        const manifest = await this.fetchManifest()
-
-        // Determine version differences
-        this.currentVersion = await this.determineCurrentVersion()
-        this.updatedVersion = (manifest as any).sha ? `${manifest.version}-${(manifest as any).sha}` : manifest.version
-
-        // Check if this update should be skipped
-        const reason = await this.skipUpdate()
-        if (reason) {
-          cli.action.stop(reason || 'done')
-        } else {
-          // Update using the new channel specification
-          await this.update(manifest, this.channel)
-        }
-
-        this.debug('tidy')
-        await this.tidy()
-        await this.config.runHook('update', {channel: this.channel})
+    await this.ensureClientDir()
+    this.debug(`Looking for locally installed versions at ${this.clientRoot}`)
+
+    // Do not show known non-local version folder names, bin and current.
+    const versions = fs
+    .readdirSync(this.clientRoot)
+    .filter(dirOrFile => dirOrFile !== 'bin' && dirOrFile !== 'current')
+    if (versions.length === 0)
+      throw new Error('No locally installed versions found.')
+    const matchingLocalVersions = versions
+    .filter(version => version.includes(targetVersion))
+    .sort((a, b) => semver.compare(b, a))
+
+    if (versions.includes(targetVersion) || matchingLocalVersions.length > 0) {
+      const target = versions.includes(targetVersion) ? targetVersion : matchingLocalVersions[0]
+      await this.updateToExistingVersion(target)
+      this.currentVersion = await this.determineCurrentVersion()
+      this.updatedVersion = target
+    } else if (channelUpdateRequested) {
```

Comment:
This logic block can likely cover both channel use case and explicit version. We want to make sure we call `this.setChannel` here as well. We can remove the `else if` block, as it should never be hit.